controllers/token_upload.go

```diff
@@ -48,6 +49,8 @@ func (u *TokenUploader) Handle(r *http.Request) error {
 	if err := json.NewDecoder(r.Body).Decode(data); err != nil {
 		return err
 	}
-
+	if data.AccessToken == "" {
+		return errors.NewBadRequest("access token can't be omitted or empty")
```

Comment:
I think this is ok as long as it ends up in the expected HTTP response with no additional headers that would make the callers think this might be coming from the kubernetes API.